### PR TITLE
[WIP] Don't Merge MemoryMarshal

### PR DESCRIPTION
**Pull request by Drawaes**
Just a start to check approach is correct.

### CODE REVIEW

**Comment by Drawaes** [comment]
For the span I can use the Unsafe methods to get the correct ref and avoid any bounds checks.

The memory is more complex, we end up with exactly the same overloads for the internal unsafe constructor and the external one. Flipping the parameters is a way to get around them unless someone else has a better idea?

File: src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        public static Span<T> DangerousCreateSpan<T>(T[] array, int start, int length) => new Span<T>(ref Unsafe.Add(ref Unsafe.As<byte, T>(ref array.GetRawSzArrayData()), start), length);
+
+        /// <summary>
+        /// Creates a memory instance over the range. However this does no bounds checking therefore is dangerous
+        /// </summary>
+        /// <param name="array">The target array.</param>
+        /// <param name="start">The index at which to begin the span.</param>
+        /// <param name="length">The number of items in the span.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Memory<T> DangerousCreateMemory<T>(T[] array, int start, int length) => new Memory<T>(start, length, array);
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs]: The approved APIs do not have Dangerous in names...

Comment (Drawaes) [reply src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs]: I forgot that got removed.

Comment (jkotas) [reply src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs]: Actually, I do not see APIs like these in the approved list. Where these exact APIs got approved?

Comment (Drawaes) [reply src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs]: https://github.com/dotnet/corefx/issues/24296

Comment (Drawaes) [reply src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs]: Basically there are two API Reviewed Issues that have been approved that converge on this single type.

Comment (jkotas) [reply src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs]: Ok, I have not noticed this API review. I do not think that the approved APIs are good.

Comment (Drawaes) [reply src/mscorlib/src/System/Runtime/InteropServices/MemoryMarshal.cs]: Okay, I will hold off until that discussion has been had over there.

**Comment by Drawaes** [comment]
Closed, as other changes are being made here and currently tied up with SslStream

**Pull request closed without merging**